src/conn/conn_dhandle.c

```diff
@@ -664,13 +664,10 @@ __conn_btree_apply_internal(WT_SESSION_IMPL *session, WT_DATA_HANDLE *dhandle,
     if (time_start != 0) {
         time_stop = __wt_clock(session);
         time_diff = WT_CLOCKDIFF_US(time_stop, time_start);
-        if (F_ISSET(S2BT(session), WT_BTREE_SKIP_CKPT)) {
-            ++conn->ckpt.handle_stats.skip;
-            conn->ckpt.handle_stats.skip_time += time_diff;
-        } else {
-            ++conn->ckpt.handle_stats.apply;
-            conn->ckpt.handle_stats.apply_time += time_diff;
-        }
+        if (F_ISSET(S2BT(session), WT_BTREE_SKIP_CKPT))
+            __wt_checkpoint_skip_handle_stats(session, &conn->ckpt, time_diff);
+        else
+            __wt_checkpoint_apply_handle_stats(session, &conn->ckpt, time_diff);
```

Comment:
Did you think to move this "if" into the function and merge the "skip" and "apply" functions into one? Then here it'd be just one function call and this logic would move inside the function.

Reply:
Yes, I pointed out that it was a bit overkill in the description above. I am glad you think the same.
The issue with this approach is that we need something from the caller to tell us if we should be in the if or else statement.

Reply:
I thought, the function could be literally like:
```c
__wt_checkpoint_apply_handle_stats(...)
{
        if (F_ISSET(S2BT(session), WT_BTREE_SKIP_CKPT)) {
...
        } else {
...
        }
```

But it's a nitpick.

Reply:
I did not want to add more btree stuff in the checkpoint but I think it should be fine, what do you think about it @sueloverso?